synapse/app/homeserver.py

```diff
@@ -620,16 +620,18 @@ def generate_user_daily_visit_stats():
     # Rather than update on per session basis, batch up the requests.
     # If you increase the loop period, the accuracy of user_daily_visits
     # table will decrease
-    clock.looping_call(generate_user_daily_visit_stats, 5 * 60 * 1000)
+    if hs.config.run_background_tasks:
+        clock.looping_call(generate_user_daily_visit_stats, 5 * 60 * 1000)
 
     # monthly active user limiting functionality
     def reap_monthly_active_users():
         return run_as_background_process(
             "reap_monthly_active_users", hs.get_datastore().reap_monthly_active_users
         )
 
-    clock.looping_call(reap_monthly_active_users, 1000 * 60 * 60)
-    reap_monthly_active_users()
+    if hs.config.run_background_tasks:
+        clock.looping_call(reap_monthly_active_users, 1000 * 60 * 60)
+        reap_monthly_active_users()
```

Comment:
This seems to invalid caches, so might not be appropriate to run on only a single instance / background worker:

https://github.com/matrix-org/synapse/blob/8a4a4186ded34bab1ffb4ee1cebcb476890da207/synapse/storage/databases/main/monthly_active_users.py#L236-L246

Reply:
`_and_stream` means that other workers will get poked about the cache invaldiations